dhash/module.py

```diff
@@ -18,20 +21,36 @@
 LIMIT_SOFT = 14
 
 MAX_ATTACHMENT_SIZE = 8000
+ALLOWED_FORMATS = ("jpg", "jpeg", "png", "webp", "gif")
+
+HEADERS = {
+    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.71 Safari/537.36"
+}
+
+URL_REGEX = r"(https?://[^\s]+)"
 
 
 class Dhash(commands.Cog):
     def __init__(self, bot):
         self.bot = bot
         self.embed_cache = {}
 
+        self.allowed_urls = HashConfig.get("allowed_urls", None)
+
+        try:
+            self.allowed_urls = re.compile(self.allowed_urls)
+        except (re.error, TypeError):
+            self.allowed_urls = None
+
     def _in_repost_channel(self, message: nextcord.Message) -> bool:
         if message.guild is None:
             return False
-        if message.attachments is None or not len(message.attachments):
-            return False
         if message.author.bot:
             return False
+        if (
+            message.attachments is None or not len(message.attachments)
+        ) and not re.search(URL_REGEX, message.content):
```

Comment:
I'd like to see Discord domains whitelisted by default. @czechbol?

Reply:
Agreed

Reply:
This would mess up whole thing, since I would have to check if string is None or "".

Reply:
I've found a way to do this by adding init funciton to DB, which initialises key to default value if it's not existing. However if you use !dhash regex unset, this value will be deleted!

Reply:
That is really user unfriendly. If there truly isn't another way, please add 'dhash regex get' function so the admin can (at least) display the defaults/the current state.

Reply:
But there's always a way -- you can check if the URL points to discord and if it does not, make the regex check ;)